### ISSUE TITLE
The appimage bundle does not work

### ISSUE THREAD

**Issue by lopippo** [xurei/hyperkeys#11]
**Environment**
- Operating system: Debian Testing
- Hyperkeys version: HyperKeys-1.1.10.AppImage

**Describe the bug and how to reproduce**
chmod a+x HyperKeys-1.1.10.AppImage
./HyperKeys-1.1.10.AppImage

the program seems to want to start, but nothing happens. This is the output:

getPath /tmp/.mount_HyperKDCCSkm/hyperkeys
getDataPath /home/rusconi/.config/hyperkeys
/tmp/.mount_HyperKDCCSkm/resources/app/icon.png

and it stalls.

**Expected behavior**
Something should happen, for example, check if the config file is there and just work or if the config file is not there open the config dialog.

**Config file**
Please provide the `macros.json` file, found at `%APPDATA%\hyperkeys\storage` on windows, `~/.config/hyperkeys/storage` on Linux.

That file does not exist.

I went to the directory where the appimage was mounted and indeed find the stuff there. Then I tried to run the hyperkeys binary directly from that directory and the output was analog to the one mentioned above. Same when running AppRun from the same mounted directory.

I will provide any details if necessary.

Thank you,
lopippo

**Comment by xurei** [comment]
Thank you for you interest for Hyperkeys.

What is the desktop environment that you are using ? 
The app icon should be visible in the notification area. It is however possible that it is not working properly on some environments.
It has been tested on GNOME2 and GNOME3

**Comment by lopippo** [comment]
> Thank you for you interest for Hyperkeys.
> 
> What is the desktop environment that you are using ?
> The app icon should be visible in the notification area. It is however possible that it is not working properly on some environments.
> It has been tested on GNOME2 and GNOME3

I am running GNOME3, package version: 3.38.

**Comment by xurei** [comment]
Weird... I'm using GNOME 3.38 as well. It has been initially created for this environment.
Could you send your GNOME plugins list ? Maybe one is clashing with it.

**Comment by xurei** [comment]
I created a fresh GNOME3 session under Manjaro and the notification icon appears on the top-right.
Please tell me if it's not the case under Debian.

![issue_11](https://user-images.githubusercontent.com/621695/106018847-43a8f400-60c2-11eb-8d15-c24f98302eca.png)

**Comment by lopippo** [comment]
> I created a fresh GNOME3 session under Manjaro and the notification icon appears on the top-right.
> Please tell me if it's not the case under Debian.
> 
> ![issue_11](https://user-images.githubusercontent.com/621695/106018847-43a8f400-60c2-11eb-8d15-c24f98302eca.png)

A screen shot to illustrate the situation :-)
![hyperkeys-appimage-fails](https://user-images.githubusercontent.com/16020903/106121670-19097a80-6158-11eb-9d9d-ad03fd10bc84.png)

**Comment by xurei** [comment]
I know that the notification area is somewhat unstable in GNOME... I believe this is the root cause. 
For a quick fix : you *might* find a solution using a plugin for the notification area. 
In the meantime, I'll take a look to find a solution in your type of environments :-)

**Comment by xurei** [comment]
I found a quick fix :
You need to install `libappindicator3-1` to get the tray icon :
```
sudo apt-get install libappindicator3-1
```

I tried it on a fresh Debian stable with success.

**Comment by lopippo** [comment]
Thank you for your endeavours,

I tried this on my testing box:

# apt search libappindicator
Sorting... Done
Full Text Search... Done
libappindicator1/now 0.4.92-8 amd64 [installed,local]
  allow applications to export a menu into the panel
libappindicator3-0.1-cil/testing 12.10.0+git20151221-5.1 amd64
  CLI binding for libayatana-appindicator3 using GObject introspection
libappindicator3-0.1-cil-dev/testing 12.10.0+git20151221-5.1 amd64
  CLI binding for libayatana-appindicator3 using GObject introspection
libghc-status-notifier-item-dev/testing 0.3.0.5-1+b3 amd64
  StatusNotifierItem/libappindicator dbus implementation
libghc-status-notifier-item-doc/testing 0.3.0.5-1 all
  StatusNotifierItem/libappindicator dbus implementation; documentation
libghc-status-notifier-item-prof/testing 0.3.0.5-1+b3 amd64
  StatusNotifierItem/libappindicator dbus implementation; profiling libraries

So it seems it is not available with version 3 in debian testing at the moment. I looked in debian sid and indeed it is there. So I'll wait for it to percolate to testing and I'll let you know. Maybe that is the library that you should include in the AppImage bundle ?

Thank you again for your research.

**Comment by xurei** [comment]
"Maybe that is the library that you should include in the AppImage bundle ?"
I'm currently trying that, but to be honest I don't know exactly how to do it...

You can also try to use `libindicator1` which is referenced in the electron docs. It didn't work on my test, though.

**Comment by lopippo** [comment]
You may try to copy it directly in the AppDir/usr/lib/x86_64-linux-gnu directory. Or in the  AppDir/usr/lib/ directory ?

**Comment by xurei** [comment]
I think I figured it out.

Could you try this version ?

https://github.com/xurei/hyperkeys/releases/download/v1.1.11-rc1/HyperKeys-1.1.11.AppImage

**Comment by lopippo** [comment]
I am sorry to have to tell you that it did not change anything. What is odd, is that the binary that is run does not seem to complain about a missing library...

**Comment by xurei** [comment]
Apparently it works if you have `gnome-shell-extension-appindicator` installed.
You can also install it from GNOME extensions page: https://extensions.gnome.org/extension/615/appindicator-support/

I keep searching ;-)

**Comment by lopippo** [comment]
Well, yes :-) and no :-(
Indeed, installing that extension has make one software that I use appear in the notification bar (so the extension works). However, the HK icon did not show up there... Sorry.

**Comment by xurei** [comment]
Have you tried with the APT package or the gnome extension website ?
I'm not sure, but I think the APT does not work indeed :-\

**Comment by xurei** [comment]
I am not sure I will be able to make it as seamless as I want :-(
There is a [long issue](https://github.com/electron/electron/issues/10427) on Electron about it. 

So I think the best path from here would be to show the main window if the extension is not installed, with a link to the extension.
Not the best solution, but given the number of comments on the main thread, I think it would at least allow you to try it.

What's your opinion about this ?

**Comment by lopippo** [comment]
> Have you tried with the APT package or the gnome extension website ?
> I'm not sure, but I think the APT does not work indeed :-\

No, I did install the extension using the browser mechanism (not apt install).

**Comment by lopippo** [comment]
> I am not sure I will be able to make it as seamless as I want :-(
> There is a [long issue](https://github.com/electron/electron/issues/10427) on Electron about it.
> 
> So I think the best path from here would be to show the main window if the extension is not installed, with a link to the extension.
> Not the best solution, but given the number of comments on the main thread, I don't think I will find a magical solution where Slack and Discord didn't.

Yeah, I think this is by far the best option. Indeed, who cares for the icon in the tray if the functionality is actually there, sitting in front of you in all its glory in a desktop window? 
> 
> It would at least allow you to try the app.
Exactly !

Then, at least, you would get more users to try the app and maybe help you iron out the remaining glitches....

Good idea. Eager to test a new version soon !
Cheers
> What's your opinion about this ?
